### PR TITLE
1510 - Test Timeouts

### PR DESCRIPTION
**Pull request by nilz3ro**
I've added some tests and described behavior for `Timeouts`.
closes casper-network/casper-node#1510

### CODE REVIEW

**Review by darthsiroftardis — APPROVED** [review]
LGTM! However, I'd wait for another approval before merging

File: node/src/components/gossiper/gossip_table.rs
Comment (Fraser999) [inline node/src/components/gossiper/gossip_table.rs:1015]: I think the test should go further, and prove that advancing the clock and further purging eventually does clear out the entries.

File: node/src/components/gossiper/gossip_table.rs
Comment (Fraser999) [inline node/src/components/gossiper/gossip_table.rs:1050]: As above, we should document that this collection purges in this way due to the implementation details of `binary_search_by`: in this case the search hits upon the `(now, 0)` entry whereas in the previous test it didn't.

Also, we should either complete the test by showing that further purges clear out the collection, or maybe just showing that the remaining entries are sorted is enough here.  Your call.

**Review by Fraser999 — COMMENTED** [review]
A couple of suggestions, but generally looking good.

File: node/src/components/gossiper/gossip_table.rs
Comment (nilz3ro) [reply node/src/components/gossiper/gossip_table.rs:1015]: I completely agree. I will update the PR soon.

File: node/src/components/gossiper/gossip_table.rs
Comment (nilz3ro) [reply node/src/components/gossiper/gossip_table.rs:1007]: Is there a way to have `cargo fmt` do this?

Comment (nilz3ro) [reply node/src/components/gossiper/gossip_table.rs:1007]: I will add a 100 char ruler to my editor for the future but I'd like to leave this as it is unless you feel strongly about it

**Review by Fraser999 — APPROVED** [review]
LGTM

**Comment by nilz3ro** [comment]
bors merge

**Pull request merged**
